### DISCUSSION TITLE
Table is missing (new case)

### DISCUSSION THREAD

**Discussion by DessisInformatica** [serverpod/serverpod#1923] [General]
I made changes to the class and table names.
Now, when running the server (dart bin/main.dart --apply-migrations), i receive warning messages, stating that the tables are missing.
How can I fix it?
I've already tried running Serverpod --create-repair and dart bin/main.dart --apply-migrations, but no success.

**Comment by Isakdl** [comment]
The integrity check is based on the generated code.

You have to run `serverpod generate` when making changes to model files to get the changes into the dart code.

To create a repair migration the correct commands are:
`serverpod create-repair-migration` and `dart bin/main.dart --apply-repair-migration`

But that shouldn't be needed for you I believe.

To create a normal migration you do:
`serverpod create-migration` and incase it is a destructive command (such as renaming a table) you have to add the flag `--force`, e.g. `serverpod create-migration --force`.

**Reply by DessisInformatica** [reply]
Hello Isak, I actually did these procedures, but without success.

**Reply by DessisInformatica** [reply]
Have any way to restart database creation?

**Reply by Isakdl** [reply]
You can wipe the database and restart from scratch if you do not care about the data. For example with the docker containers you get for development you can run `docker compose down -v` to delete the images and volumes (including data and table definitions).

Other than this the repair migration flow is intended to be able to get to a valid state in the db.

**Reply by Lord-Platypus** [reply]
I had the same problem and I tried your solution, it didn't work at first. 

Serverpod was unable to see the changes in my models, so I did what any flutter developer would do: `flutter clean` and `flutter pub get`, then I tried with `serverpod generate` and `serverpod create-migration`

After that the new migration command gave me the warnings about the old tables being dropped, so it was fixed.

Hope this helps someone

**Comment by DessisInformatica** [comment]
I got it. Thanks.
